core/src/main/java/com/halcyonmobile/android/core/internal/usecase/GetGitHubReposPaginatedUseCase.kt

```diff
@@ -1,25 +1,27 @@
 package com.halcyonmobile.android.core.internal.usecase
 
-import com.halcyonmobile.android.core.internal.repo.GitHubRepoRepository
+import com.halcyonmobile.android.core.ErrorLogger
+import com.halcyonmobile.android.core.internal.repo.GithubRepoRepositoryHelper
 import com.halcyonmobile.android.core.model.GitHubRepo
 import com.halcyonmobile.android.paging.PagingStateMachine
 import kotlinx.coroutines.flow.Flow
 
-class GetGitHubReposPaginatedUseCase internal constructor(private val gitHubRepoRepository: GitHubRepoRepository) {
+class GetGitHubReposPaginatedUseCase internal constructor(private val gitHubRepoGithubRepoRepository: GithubRepoRepositoryHelper<GitHubRepo>) {
 
-    operator fun invoke(pageSize: Int = 10) : PagingStateMachine<GitHubRepo> {
-        return PagingStateMachine(pageSize = pageSize, requestElements =  { numberOfElements, fetch ->
-            wrapRepoResponse {
-                if (fetch) gitHubRepoRepository.fetch(numberOfElements) else gitHubRepoRepository.get(numberOfElements)
+    operator fun invoke(pageSize: Int = 10): PagingStateMachine<GitHubRepo> {
+        return PagingStateMachine(pageSize = pageSize, requestElements = { numberOfElements, fetch ->
+            wrapRepoResponse(gitHubRepoGithubRepoRepository.errorLogger) {
```

Comment:
Something like this would be also better to be done via the helper class imo, maybe make it open so if not exceptions are used for error handling it's still possible to use